plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -547,24 +571,30 @@ class producer_plugin_impl : public std::enable_shared_from_this<producer_plugin
             }
 
             if (except_ptr) {
+               auto [contract_name, act_name, details] = get_detailed_contract_exception_info(except_ptr, response);
```

Comment:
Thanks for the comment to avoid unnecessary call. With returning `fc::mutable_variant_object` within the log statement, does the following output look good?
`debug 2022-08-10T14:38:29.504 nodeos    producer_plugin.cpp:581       operator()           ] [TRX_TRACE] Block 10058 for producer eosio is REJECTING tx: 1b43cf20688149c8b1d42f1603ab20c0bd35206b37923eacc43b8791376741f8, auth: eosio, {"contract":"hello","action":"hi","details":"assertion failure with message: This is a test"}`

Reply:
Will change `details` above to `message`.

Reply:
I personally do not like that example output. The embedded JSON seems rather verbose to me.

Reply:
I just made `get_detailed_contract_exception_info` to return a string. An example log would look like

`debug 2022-08-10T21:56:12.249 nodeos    producer_plugin.cpp:575       operator()           ] [TRX_TRACE] Block 10294 for producer eosio is REJECTING tx: d56f2f044ef34155958919983d2532df30c7e9c3a7fa8cce0f465992b7a8a6f2, auth: eosio, action::hello:hi:assertion failure with message: This is a test`

Reply:
Should there be space after "action: "?  Why is there double "::"?  
The 'with' is not adding value?
Is there always a message, or message is optional?

What if the action is "assertion failure with message:" (just making stupid example). Is that even valid? Then the log would be:

`assertion failure with message: with message: This is a test`

This would get parsed wrong probably.

We have two strings of arbitrary length making it difficult to deal with.  

`debug 2022-08-10T21:56:12.249 nodeos producer_plugin.cpp:575 operator() ] [TRX_TRACE] Block 10294 for producer eosio is REJECTING tx: d56f2f044ef34155958919983d2532df30c7e9c3a7fa8cce0f465992b7a8a6f2, auth: eosio, action::hello:hi:assertion failure // This is a test`

Would be easier to parse, assuming '//' is not valid 'action'.

Reply:
(Replace // with any other unique separator)

Reply:
"assertion failure with message:" is added by host function eosio_assert or eosio_assert_message, which is better to be left alone. How about `action: hello:hi, assertion failure with message: This is a test`?

Reply:
What are the valid characters for "action"?  Is comma never used?  If yes, then this is fine because I can depend on it to parse the message (before comma and after comma).

I don't know the nodeos internals, I am maybe not making reasonable suggestions.

Reply:
action's type is eosio::name, which can only be 'a' to 'z' and '1' to '5'. We are good here. Will push a change after I am done with some other work.